taichi/backends/opengl/codegen_opengl.cpp

```diff
@@ -81,9 +71,11 @@ class KernelGen : public IRVisitor {
   std::string root_snode_type_name_;
   std::string glsl_kernel_prefix_;
   int glsl_kernel_count_{0};
-  int num_threads_{1};
-  int num_groups_{1};
-  RangeSizeEvaluator range_size_evaluator_;
+
+  bool is_top_level_{true};
+  LineAppender line_appender_, line_appender_header_;
```

Comment:
```suggestion
  LineAppender line_appender_;
  LineAppender line_appender_header_;
```

Reply:
I'm fine with this, just ask if this is a suggested C++ coding style to never use `,` when declaring variables?

Reply:
It can be misused, e.g. when you declare an object and its pointer type in the same line. Also see https://wiki.sei.cmu.edu/confluence/display/c/DCL04-C.+Do+not+declare+more+than+one+variable+per+declaration . Personally I think this is controversial, so the simplest rule is just not to use it.